### PR TITLE
[SPIR-V] change fixing inconsistent terminators to asserts

### PR DESCRIPTION
**Pull request by iliya-diyachkov**
It's to test SPIRVBlockLabeler without fixing inconsistent terminators by OpenCL CTS.

### CODE REVIEW

File: llvm/lib/Target/SPIRV/SPIRVBlockLabeler.cpp [hunk 1]
```diff
@@ -116,25 +113,15 @@ bool SPIRVBlockLabeler::runOnMachineFunction(MachineFunction &MF) {
       if (MI.getOpcode() == SPIRV::OpBranch) {
         Branches.push_back(&MI);
       } else if (MI.getOpcode() == SPIRV::OpBranchConditional) {
-        // If previous optimizations generated conditionals with implicit
-        // fallthrough, add the missing explicit "else" to it valid SPIR-V
-        if (MI.getNumOperands() < 3) {
-          MI.addOperand(MachineOperand::CreateMBB(MBB.getNextNode()));
-        }
+        assert(MI.getNumOperands() == 3 &&
```
Comment (zuban32) [inline llvm/lib/Target/SPIRV/SPIRVBlockLabeler.cpp:116]: The spec says it's possible we may have >3 operands for OpBranchCond, I agree we're currently processing only 3 of them though

**Comment by zuban32** [comment]
Does any of our LITs fail with this? I'd suspect some Verifier problems

File: llvm/lib/Target/SPIRV/SPIRVBlockLabeler.cpp (hunk 1 above)
Comment (iliya-diyachkov) [reply llvm/lib/Target/SPIRV/SPIRVBlockLabeler.cpp:116]: Yes, we don't create OpBranchConditional with 5 operands (3 normal+2 branch weights). We can add TODO comment about this spec case here.

**Comment by iliya-diyachkov** [comment]
Neither LITs nor CTS tests (which passes without the patch) fail with this patch. I have found only one issue on transcoding/OpSwitchEmpty.ll test which did not pass before with "wrong output" and got compile fail with this patch. It failed with assertion "Missed terminator in MBB" in BB which is a header of empty switch instruction. Taking into account that we still have no proper switch support, I think this is the issue of previous passes that they do not inserted a branch instruction in that BB.

**Comment by zuban32** [comment]
I see. Although I wouldn't be so sure about safety of this removal - our branches currently don't comply with `MachineVerifier` at all - they're not marked as barriers and they're operating over MBBs directly, so `MachineVerifier` fails (if enabled, e.g. put in the very beginning of `addPreEmitPass2`) even before this labelling pass and branches seem to be the only problem.

I think the reasonable assumption would be that the code should pass verification on all the stages before the globals hoisting which breaks everything inevitably and by-design still.

**Comment by iliya-diyachkov** [comment]
The patch was verified for LLVM base implementation. I do not insist to put it in our branch right now. On other hand the patch does not break any test so it's safe enough to apply it. Moreover the pass with these asserts works as a branch verifier.

**Comment by zuban32** [comment]
No problem, I have no objections. Will try to take a look at the branches verification to see if it may be worth fixing

**Pull request merged**